src/message/connection.js

```diff
@@ -531,7 +531,8 @@ Connection.prototype._tryReconnect = function () {
   } else {
     this._clearReconnect()
     this.close()
-    this._client.emit(C.MAX_RECONNECTION_ATTEMPTS_REACHED, this._reconnectionAttempt)
+    this._setState(C.CONNECTION_STATE.CLOSED)
```

Comment:
i've just removed the setState.
i'll check again final MAX_RECONNECTION_ATTEMPTS_REACHED connection state and let you know. 
thanks